tests/ui/print_type_sizes/async.stdout

```diff
@@ -1,4 +1,4 @@
-print-type-size type: `[async fn body@$DIR/async.rs:8:36: 11:2]`: 16386 bytes, alignment: 1 bytes
+print-type-size type: `Generator(DefId(0:7 ~ async[505d]::test::{closure#0}), [Adt(std::future::ResumeTy, []), Tuple([]), Tuple([]), GeneratorWitness(Binder([Adt(std::future::ResumeTy, []), Array(Uint(U8), Const { ty: Uint(Usize), kind: Value(Leaf(0x0000000000002000)) }), Generator(DefId(0:4 ~ async[505d]::wait::{closure#0}), [Adt(std::future::ResumeTy, []), Tuple([]), Tuple([]), GeneratorWitness(Binder([], [])), Tuple([])], Static), Tuple([])], [])), Tuple([Array(Uint(U8), Const { ty: Uint(Usize), kind: Value(Leaf(0x0000000000002000)) })])], Static)`: 16386 bytes, alignment: 1 bytes
```

Comment:
This is basically unreadable, and used by a lot of end-users. Can you make this use the display impl for types?